common/src/main/java/com/microsoft/identity/common/internal/broker/PackageHelper.java

```diff
@@ -160,4 +159,53 @@ public static String getBrokerRedirectUrl(final String packageName, final String
         }
         return "";
     }
+
+    /**
+     * Helper method to get signatures in a back-compatible way
+     *
+     * @param packageInfo A packageInfo instance with the flag PackageManager.GET_SIGNING_CERTIFICATES/PackageManager.GET_SIGNATURES set
+     * @return Signature[] or null
+     */
+    public static Signature[] getSignatures(final PackageInfo packageInfo) {
+        if (packageInfo == null) {
+            return null;
+        }
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+            if (packageInfo.signingInfo == null) {
+                return null;
+            }
+            if (packageInfo.signingInfo.hasMultipleSigners()) {
+                return packageInfo.signingInfo.getApkContentsSigners();
+            } else {
+                return packageInfo.signingInfo.getSigningCertificateHistory();
+            }
+        }
+
+        return packageInfo.signatures;
+    }
+
+    public static int getPackageManagerFlag() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+            return PackageManager.GET_SIGNING_CERTIFICATES;
+        }
+
+        return PackageManager.GET_SIGNATURES;
+    }
+
+    public static PackageInfo getPackageInfo(@NonNull PackageManager packageManager, @NonNull String packageName) throws PackageManager.NameNotFoundException {
+        return packageManager.getPackageInfo(packageName, getPackageManagerFlag());
+    }
+
+    public static PackageInfo getPackageInfo(@NonNull Context context, @NonNull String packageName) throws PackageManager.NameNotFoundException {
```

Comment:
Let's get rid of this method and instead force callers to use the following overload

```java
getPackageInfo(@NonNull PackageManager packageManager, @NonNull String packageName)
```

No apparent need I can see to [pass the entire context](https://en.wikipedia.org/wiki/Principle_of_least_privilege)